lib/database_validations/rubocop/cop/belongs_to.rb

```diff
@@ -0,0 +1,46 @@
+module RuboCop
+  module Cop
+    module DatabaseValidations
+      # Use `db_belongs`_to instead of `belongs_to`.
+      #
+      # @example
+      #   # bad
+      #   belongs_to :company
+      #
+      #   # good
+      #   db_belongs_to :company
+      #
+      class BelongsTo < Cop
+        MSG = 'Use `db_belongs_to`.'.freeze
+
+        NON_SUPPORTED_OPTIONS = %i[
+          optional
+          required
+          polymorphic
+          foreign_type
+        ].freeze
+
+        def_node_matcher :belongs_to?, '(send nil? :belongs_to ...)'
```

Comment:
I guess we can join the expressions using `[]` and combine them with `#supported_types`:

```suggestion
        def_node_matcher :offensor?, '(send nil? :belongs_to (args (hash (pair (sym #not_supported)))'
```

and then supported_arguments is pretty similar to `supported` but it's already receiving the args:

```ruby
     def  not_supported(option)
        NON_SUPPORTED_OPTIONS.include?(option)
     end
```

and `on_send` becomes positive:

```ruby
     add_offense(node, location: :selector) if offensor?(node)
```
and we can drop the `option_name` too :)

I didn't tested but I guess the intent is clear.

Reply:
Your matcher won't work because there may be other pairs and order is not guaranteed. It's a missing feature on rubocop to be able to match on any pair.